pixl_core/src/core/project_config/secrets.py

```diff
@@ -71,7 +84,7 @@ def _check_system_envvar(var_name: str) -> None:
 
 
 @lru_cache
-def _fetch_secret(secret_name: str) -> str:
+def _fetch_secret(client: SecretClient, secret_name: str) -> str:
```

Comment:
see below for all arguments having to be hashable. check locally that the caching is truly being used